angularSTEP/src/app/recipe-page/recipe-page.component.ts

```diff
@@ -13,17 +18,128 @@ import {RecipeConverter} from '../recipe-converter';
 export class RecipePageComponent {
   id: string | null = this.route.snapshot.paramMap.get('id');
   pageRecipe!: Recipe;
+  baseRecipe!: Recipe;
+  currentUser!: User;
+  baseUser!: User;
+  loggedIn: boolean = false;
 
-  constructor(private db: AngularFirestore,
-              private route: ActivatedRoute,) {
-                this.setRecipeData();
+  constructor(
+    private db: AngularFirestore,
+    private route: ActivatedRoute,
+    private fAuth: AngularFireAuth,
+    private dialog: MatDialog,
+    private router: Router,
+  ) {
+    this.setPageData();
+    this.fAuth.onAuthStateChanged(auth => {
+      if (auth) {
+        this.setPageData();
+        this.setUserData(auth.uid);
+        this.loggedIn = true;
+      } else {
+        this.setPageData();
+        this.loggedIn = false;
+      }
+    });
   }
 
-  async setRecipeData() {
-    const postRecipe = await this.db.doc('/recipes/' + this.id + '/').ref.withConverter(new RecipeConverter().recipeConverter).get();
+  async setPageData() {
+    const postRecipe = await this.db.doc('/recipes/' + this.id + '/')
+      .ref.withConverter(new RecipeConverter().recipeConverter).get();
     const recipeClassData = postRecipe.data();
+
     if (recipeClassData) {
+
+
+      if (recipeClassData.baseUploaderUid) {
+        const postBaseUser = await this.db.doc('/users/' + recipeClassData.baseUploaderUid + '/')
```

Comment:
postBaseuser can be null here, so accessing data() can be not good. I would add an if statement